bindings/wasm/examples/src/explorer-did.js

```diff
@@ -0,0 +1,243 @@
+
+    // Publish the diff to the Tangle.
+    // Note that we still use the `messageId` from the last integration chain message here to link
+    // the current diff chain to that point on the integration chain.
+    const diffReceipt2 = await client.publishDiff(intReceipt2.messageId, diff2);
+
+
+
+    // ===========================================================================
+    // Diff Chain Update 3
+    // ===========================================================================
+
+    console.log("start diff update 3");
+
+    // Prepare another diff chain update.
+    const diffDoc3 = Document.fromJSON(diffDoc2.toJSON());
+
+    // Add a second Service with the tag "linked-domain-2"
+    let serviceJSON3 = {
+        id: diffDoc3.id + "#linked-domain-3",
+        type: "LinkedDomains",
+        serviceEndpoint: {
+            "origins": ["https://iota.org/", "https://example.com/"]
+        },
+    };
+    diffDoc3.insertService(Service.fromJSON(serviceJSON3));
+    diffDoc3.metadataUpdated = Timestamp.nowUTC();
+
+    // This is the second diff therefore its `previousMessageId` property is
+    // set to the first published diff to extend the diff chain.
+    const diff3 = diffDoc2.diff(diffDoc3, diffReceipt2.messageId, key, diffDoc2.defaultSigningMethod().id.toString());
+
+    // Publish the diff to the Tangle.
+    // Note that we still use the `messageId` from the last integration chain message here to link
+    // the current diff chain to that point on the integration chain.
+    const diffReceipt3 = await client.publishDiff(intReceipt2.messageId, diff3);
+
+
+    // ===========================================================================
+    // Integration Chain Update 3
+    // ===========================================================================
+    console.log("start integration update 3");
+
+    // Publish a second integration chain update
+    let intDoc3 = Document.fromJSON(diffDoc3.toJSON());
+
+    // Remove the #keys-1 VerificationMethod
+    intDoc3.removeMethod(intDoc3.id.toUrl().join("#keys-1"));
+
+    // Remove the #linked-domain-1 Service
+    intDoc3.removeService(intDoc3.id.toUrl().join("#linked-domain-1"));
+
+    // Add a VerificationMethod with a new KeyPair, called "keys-2"
+    // const keys3 = new KeyPair(KeyType.Ed25519);
+    // const method3 = VerificationMethod.fromDID(intDoc2.id, keys2, "keys-2");
+    // intDoc2.insertMethod(method2, MethodScope.VerificationMethod());
+
+    // Note: the `previous_message_id` points to the `message_id` of the last integration chain
+    //       update, NOT the last diff chain message.
+    intDoc2.metadataPreviousMessageId = intReceipt2.messageId;
+    intDoc2.metadataUpdated = Timestamp.nowUTC();
+    intDoc2.signSelf(key, intDoc2.defaultSigningMethod().id.toString());
+    const intReceipt3 = await client.publishDocument(intDoc2);
```

Comment:
This publishes the wrong document.

```suggestion
    // Note: the `previous_message_id` points to the `message_id` of the last integration chain
    //       update, NOT the last diff chain message.
    intDoc3.metadataPreviousMessageId = intReceipt2.messageId;
    intDoc3.metadataUpdated = Timestamp.nowUTC();
    intDoc3.signSelf(key, intDoc3.defaultSigningMethod().id.toString());
    const intReceipt3 = await client.publishDocument(intDoc3);
```